test/unit/test_grouped_loader.py

```diff
@@ -0,0 +1,238 @@
+"""
+Tests for deepreg/dataset/loader/grouped_loader.py in
+pytest style
+"""
+from os.path import join
+
+import numpy as np
+import pytest
+
+from deepreg.dataset.loader.grouped_loader import GroupedDataLoader
+from deepreg.dataset.loader.h5_loader import H5FileLoader
+from deepreg.dataset.loader.nifti_loader import NiftiFileLoader
+
+FileLoaderDict = dict(nifti=NiftiFileLoader, h5=H5FileLoader)
+nifti_path = join("data", "test", "nifti", "grouped")
+h5_path = join("data", "test", "h5", "grouped")
+DataPaths = dict(nifti=nifti_path, h5=h5_path)
+
+
+def test_validate_data_files():
+    """
+    Test validate_data_files function looks for inconsistencies in the fixed/moving image and label lists.
+    If there is any issue it will raise an error, otherwise it returns None.
+    """
+    for key_file_loader, file_loader in FileLoaderDict.items():
+        for train_split in ["train", "test"]:
+            for labeled in ["True", "False", 1, 0]:
+                data_dir_path = join(DataPaths[key_file_loader], train_split)
+                image_shape = (64, 64, 60)
+                common_args = dict(
+                    file_loader=file_loader,
+                    labeled=labeled,
+                    sample_label="all",
+                    intra_group_prob=1,
+                    intra_group_option="forward",
+                    sample_image_in_group=False,
+                    seed=None if train_split == "train" else 0,
+                )
+
+                data_loader = GroupedDataLoader(
+                    data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+                )
+
+                assert data_loader.validate_data_files() is None
+
+
+def test_num_intra_group_probs():
+    """
+    Test whether fewer than two groups throws an error for intra_group_prob < 1
+     - data/test/h5/grouped/test contains only a single group
+    """
+    with pytest.raises(Exception) as e_info:
+        data_dir_path = join(DataPaths["h5"], "test")
+        image_shape = (64, 64, 60)
+        common_args = dict(
+            file_loader=H5FileLoader,
+            labeled=True,
+            sample_label="all",
+            intra_group_prob=0.5,
+            intra_group_option="forward",
+            sample_image_in_group=False,
+            seed=0,
+        )
+        data_loader = GroupedDataLoader(
+            data_dir_path=data_dir_path, image_shape=image_shape, **common_args
+        )
+        data_loader.close()
+
+        assert "we need at least two groups" in str(e_info.value)
+
+
+def test_sample_indices_intra():
+    """
+    Test the number of samples and indices are correct for a single image sample
```

Comment:
this is not single image sample case?

if I understand the code correctly, we are testing "sample image mode" here, and therefore it's independent of intra or inter.

also, this is part of the test for init.

but maybe we should define another function as we have a block of the code defining `sample_indices` ? what do you think?

Reply:
My mistake, incorporated test function into `test_init()`